src/main/java/nl/novi/techiteasy/services/TelevisionServiceImpl.java

```diff
@@ -0,0 +1,180 @@
+package nl.novi.techiteasy.services;
+
+import nl.novi.techiteasy.dtos.TelevisionDto;
+import nl.novi.techiteasy.dtos.TelevisionInputDto;
+import nl.novi.techiteasy.exceptions.RecordNotFoundException;
+import nl.novi.techiteasy.models.Television;
+import nl.novi.techiteasy.repositories.TelevisionRepository;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class TelevisionServiceImpl implements TelevisionService {
+    private final TelevisionRepository televisionRepository;
+
+    public TelevisionServiceImpl(TelevisionRepository televisionRepository) {
+        this.televisionRepository = televisionRepository;
+    }
+
+    @Override
+    public List<TelevisionDto> getTelevisions() {
+        List televisionList = televisionRepository.findAll();
+        if (emptyList(televisionList)) {
+            throw new RecordNotFoundException("No televisions in the database");
+        } else {
+            return addToList(televisionList);
+        }
+    }
+    @Override
+    public TelevisionDto getTelevision(Long id) {
+        if (idPresent(id)) {
+            return fromTelevision(televisionRepository.findById(id).get());
+        } else {
+            throw new RecordNotFoundException("This television with id " + id + " doesn't exist.");
+        }
+    }
+    @Override
+    public TelevisionDto saveTelevision(TelevisionInputDto television) {
+        Television tv = toTelevision(television);
+        televisionRepository.save(tv);
+        TelevisionDto savedTv = fromTelevision(tv);
+
+        return savedTv;
+    }
+    @Override
+    public TelevisionDto updateTelevision(TelevisionInputDto updateTelevision, Long id) {
+        if (idPresent(id)) {
+           Television tv = saveChanges(id, updateTelevision);
+            return fromTelevision(tv);
+        } else {
+            throw new RecordNotFoundException("Could not find tv with id: " + id);
+        }
+    }
+    @Override
+    public void deleteTelevision(Long id) {
+        try {
+            televisionRepository.deleteById(id);
+        } catch (EmptyResultDataAccessException ex) {
+            throw new RecordNotFoundException("Cannot find television with id: " + id);
+        }
+    }
+    @Override
+    public Television saveChanges(Long id, TelevisionInputDto updateTelevision) {
+        Television tv = televisionRepository.findById(id).get();
```

Comment:
Ik heb vandaag nog een callbackfunctie in mijn code bijgeschreven. Misschien voor jou een kleine leuke toevoeging :)))
Television tv = televisionRepository.findById(id).orElseThrow(() -> {throw new RecordNotFoundException("Tv with this id doesn't exist. Try another Id.");});